### PR TITLE
Rework of workspace sampling, fix goal manifold sampling, visualize frame and manifold

### PR DESCRIPTION
**Pull request by ikrets**
Workspace sampling rework: the `WorkspaceSampler` interface has changed. Now it allows for easier combination of different position and orientation sampling strategies together. Here is an example of usage:

```
WorkspaceSampler goal_manifold_sampler(
    UniformPositionInAsymmetricBox(goal_transform, req.min_position_deltas,
                                   req.max_position_deltas),
    DeltaXYZOrientation(rl::math::Quaternion(goal_transform.linear()),
                        req.min_orientation_deltas, req.max_orientation_deltas));

std::mt19937 generator(time(nullptr));
std::uniform_real_distribution<double> random_01;
auto sample_01 = [&generator, &random_01]() { return random_01(generator); };

rl::math::Transform sampled_transform = goal_manifold_sampler.generate(sample_01);
```

Modifications to the /check_kinematics service call:
1. The goal manifold is now aligned with the goal frame.
2. Visualizations in the viewer: goal frame, goal manifold (with a gray transparent box).

### CODE REVIEW

File: src/service_worker.cpp [hunk 1]
```diff
@@ -146,18 +168,18 @@ bool ServiceWorker::checkKinematicsQuery(tub_feasibility_check::CheckKinematics:
   WorldCollisionTypes world_collision_types(part_to_type);
 
   ROS_INFO("Setting ifco pose and creating bounding boxes");
-  ifco_scene->moveIfco(ifco_transform);
-  ifco_scene->removeBoxes();
+  ifco_scene_->moveIfco(ifco_transform);
```
Comment (zweistein) [inline src/service_worker.cpp]: Why do you need to change the name? This change is not required for functionality porous but increases the diff. Please revert the name.

**Comment by zweistein** [comment]
I observed a naming issue in `rlkin` and `rlsg`: the model names end with **ocado2** 
This last part of the name should refer to the end-effector, so please rename it to **RBOHand2**

File: src/service_worker.cpp (hunk 1 above)
Comment (ikrets) [reply src/service_worker.cpp]: This was addressed in 562c9c12bc4146919d4d063a9570a9e0e74d79ad.

**Comment by ikrets** [comment]
> I observed a naming issue in `rlkin` and `rlsg`: the model names end with **ocado2**
> This last part of the name should refer to the end-effector, so please rename it to **RBOHand2**

This was addressed in 562c9c12bc4146919d4d063a9570a9e0e74d79ad.

**Comment by zweistein** [comment]
I would suggest merging the branch that is tested on the real system. That should be basically the same, right?

**Comment by ikrets** [comment]
> I would suggest merging the branch that is tested on the real system. That should be basically the same, right?

Yes, let's not merge anything until there is a state that's working on the real system.

**Pull request closed without merging**